src/Spark.Engine.Test/Service/PatchServiceTests.cs

```diff
@@ -174,6 +186,57 @@ public void CanApplyCollectionAddPatchForNonNamedDataTypes()
             Assert.Equal("testProcessing", resource.Processing[0].Description);
             Assert.Equal(dateTime, resource.Processing[0].Time);
         }
+        
+        [Fact]
+        public void CanApplyCollectionAddPatchForNonNamedDataTypesWithExtension()
+        {
+            var parameters = new Parameters();
+            var extensions = new List<Extension>()
+            {
+                new Extension("http://extensions.org/extensionResourceReference", new ResourceReference("Device/1")),
+                new Extension("http://extensions.org/extensionCode", new Code("someCode")),
+                new Extension("http://extensions.org/extensionUrl", new FhirUri("someUrl")),
+                new Extension("http://extensions.org/extensionString", new FhirString("someString")),
+                new Extension("http://extensions.org/extensionDateTime", new FhirDateTime(DateTimeOffset.Now))
+            };
+            parameters.AddAddPatchParameter("Specimen", "processing", null);
+            var valuePart = parameters.Parameter[0].Part[3];
+            valuePart.Name = "value";
+            foreach (var extension in extensions)
+            {
+                valuePart.Part.Add(new Parameters.ParameterComponent()
+                {
+                    Name = "extension", Part = new List<Parameters.ParameterComponent>()
+                    {
+                        new Parameters.ParameterComponent()
+                        {
+                            Name = "url", Value = new FhirUri(extension.Url) 
+                        },
+                        new Parameters.ParameterComponent()
+                        {
+                            Name = "value", Value = extension.Value
+                        }
+                    }
+                });
+            }
```

Comment:
New extension methods could be added, so it would allow to add generic parts as a value. Like AddAddPatchParameter(string path, string name, List<Parameters.ParameterComponent> parts).
Though those are part of the Hl7.Fhir.Rest, so it would need to go through there first.